server/message/message_server/server.go

```diff
@@ -266,6 +268,74 @@ func (*server) GetMessagesByChatroom(ctx context.Context, req *messagepb.GetMess
 
 }
 
+func (*server) DeleteMessage(ctx context.Context, req *messagepb.DeleteMessageRequest) (*messagepb.DeleteMessageResponse, error) {
+	user_id, err := common_jwt.ReadUUIDFromContext(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	if ok, err := interceptors.IsUserInChatroom(req.ChatroomID, user_id); !ok || err != nil {
+		if !ok && err == nil {
+			log.Printf("User %v not found in this chatroom: %v", user_id, req.ChatroomID)
+			return nil, status.Errorf(
+				codes.Unauthenticated,
+				fmt.Sprintf("User %v not found in this chatroom: %v", user_id, req.ChatroomID),
+			)
+		}
+
+		log.Printf("Error while checking if user belong to chatroom: %v", err)
+		return nil, status.Errorf(
+			codes.Internal,
+			fmt.Sprintf("Error while checking if user belong to chatroom: %v", err),
+		)
+
+	}
+
+	//Elastic client init with Olivere package
+	client, err:= elastic.NewClient(
+		elastic.SetSniff(true),
+		elastic.SetURL("http://localhost:9200"),
+		elastic.SetHealthcheckInterval(5*time.Second),
+	)
+	if err != nil {
+		log.Printf("Elasticsearch connection error %v", err)
+		return nil, status.Errorf(
+			codes.Internal,
+			fmt.Sprintf("Elasticsearch connection error %v", err),
+		)
+	}
+
+	exist, err := client.IndexExists(req.ChatroomID).Do(ctx)
+	if !exist {
+		log.Printf("Index %v not found", req.ChatroomID)
+		return nil, status.Errorf(
+			codes.NotFound,
+			fmt.Sprintf("Index %v not found", req.ChatroomID),
+		)
+	}
+
```

Comment:
We'd need to check if the person who asks for the message deletion is actually the author of the message.

Get the message from ES => Check if `userId == authorUUID` => And then pass or reject